app/src/main/java/com/randomappsinc/studentpicker/editing/EditNameListFragment.java

```diff
@@ -120,6 +121,11 @@ public void showNameOptions(final int position) {
         nameEditChoicesDialog.showChoices(namesAdapter.getItem(position));
     }
 
+    @OnClick(R.id.voice_entry_icon)
+    public void voiceEntry() {
+        SpeechUtil.openSpeechToTextDialog(getActivity());
```

Comment:
Hmmm, I just realized that using the activity isn't great since we should use the startActivity of the fragment. This will allow us to keep the onActivityResult block in the fragment where the logic rightfully belongs. Let's change the utility method to only construct the intent.